### ISSUE TITLE
Java error running "nodetool status": Could not initialize class sun.reflect.misc.MethodUtil

### ISSUE THREAD

**Issue by wmac** [scylladb/scylla#5071]
Hello,

I am running into below error when trying to run "nodetool status". I guess it happened after I updated some server software.

Scylla version (or git commit hash): 3
Cluster size: 2
OS (RHEL/CentOS/Ubuntu/AWS AMI): Centos 7.6

---------------
```
error: Could not initialize class sun.reflect.misc.MethodUtil
-- StackTrace --
java.lang.NoClassDefFoundError: Could not initialize class sun.reflect.misc.Meth                                                                                                                     odUtil
        at com.sun.jmx.mbeanserver.StandardMBeanIntrospector.invokeM2(StandardMB                                                                                                                     eanIntrospector.java:112)
        at com.sun.jmx.mbeanserver.StandardMBeanIntrospector.invokeM2(StandardMB                                                                                                                     eanIntrospector.java:46)
        at com.sun.jmx.mbeanserver.MBeanIntrospector.invokeM(MBeanIntrospector.j                                                                                                                     ava:237)
        at com.sun.jmx.mbeanserver.PerInterface.getAttribute(PerInterface.java:8                                                                                                                     3)
        at com.sun.jmx.mbeanserver.MBeanSupport.getAttribute(MBeanSupport.java:2                                                                                                                     06)
        at com.sun.jmx.interceptor.DefaultMBeanServerInterceptor.getAttribute(De                                                                                                                     faultMBeanServerInterceptor.java:647)
        at com.sun.jmx.mbeanserver.JmxMBeanServer.getAttribute(JmxMBeanServer.ja                                                                                                                     va:678)
        at com.scylladb.jmx.utils.APIMBeanServer.getAttribute(APIMBeanServer.jav                                                                                                                     a:139)
        at javax.management.remote.rmi.RMIConnectionImpl.doOperation(RMIConnecti                                                                                                                     onImpl.java:1445)
        at javax.management.remote.rmi.RMIConnectionImpl.access$300(RMIConnectio                                                                                                                     nImpl.java:76)
        at javax.management.remote.rmi.RMIConnectionImpl$PrivilegedOperation.run                                                                                                                     (RMIConnectionImpl.java:1309)
        at javax.management.remote.rmi.RMIConnectionImpl.doPrivilegedOperation(R                                                                                                                     MIConnectionImpl.java:1401)
        at javax.management.remote.rmi.RMIConnectionImpl.getAttribute(RMIConnect                                                                                                                     ionImpl.java:639)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.                                                                                                                     java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAcces                                                                                                                     sorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at sun.rmi.server.UnicastServerRef.dispatch(UnicastServerRef.java:357)
        at sun.rmi.transport.Transport$1.run(Transport.java:200)
        at sun.rmi.transport.Transport$1.run(Transport.java:197)
        at java.security.AccessController.doPrivileged(Native Method)
        at sun.rmi.transport.Transport.serviceCall(Transport.java:196)
        at sun.rmi.transport.tcp.TCPTransport.handleMessages(TCPTransport.java:5                                                                                                                     73)
        at sun.rmi.transport.tcp.TCPTransport$ConnectionHandler.run0(TCPTranspor                                                                                                                     t.java:834)
        at sun.rmi.transport.tcp.TCPTransport$ConnectionHandler.lambda$run$0(TCP                                                                                                                     Transport.java:688)
        at java.security.AccessController.doPrivileged(Native Method)
        at sun.rmi.transport.tcp.TCPTransport$ConnectionHandler.run(TCPTransport                                                                                                                     .java:687)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.                                                                                                                     java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor                                                                                                                     .java:624)
        at java.lang.Thread.run(Thread.java:748)
```

**Comment by amoskong** [comment]
Hi wmac,

Are you using scylla 3.0? What's your java version?

You can check the dependency by `rpm -qaR scylla-tools-core`
java-1.8.0-openjdk-headless

**Comment by wmac** [comment]
Hi Amos,

Thank you for your time. Yes, I am using version 3 and openjdk 1.8

Regards

**Comment by slivne** [comment]
@wmac to try and debug this we need to check what JVM is actually ran and what is the CLASSPATH

please edit the nodetool script and just before the line 

```
"$JAVA" $JAVA_AGENT -ea -cp "$CLASSPATH" $JVM_OPTS -Xmx$MAX_HEAP_SIZE \
        -XX:ParallelGCThreads=1 \
        -Dcassandra.storagedir="$cassandra_storagedir" \
        -Dlogback.configurationFile=logback-tools.xml \
        $JVM_ARGS \
        org.apache.cassandra.tools.NodeTool -p $JMX_PORT $ARGS
```

add two lines

```
echo $JAVA
echo $CLASSPATH
```

that should help validate the environment used by the nodetool command

**Comment by wmac** [comment]
@Shlomi, Hi

Thank you very much for your time. This is the output:
Just that nodetool works but the "status" subcommand has problem.

/bin/java
/tmp/tmp.TwKLWhneft:/usr/share/scylla/cassandra/lib/airline-0.6.jar:/usr/share/s
cylla/cassandra/lib/antlr-runtime-3.5.2.jar:/usr/share/scylla/cassandra/lib/asm-
5.0.4.jar:/usr/share/scylla/cassandra/lib/caffeine-2.2.6.jar:/usr/share/scylla/c
assandra/lib/cassandra-driver-core-3.5.1-shaded.jar:/usr/share/scylla/cassandra/
lib/commons-cli-1.1.jar:/usr/share/scylla/cassandra/lib/commons-codec-1.9.jar:/u
sr/share/scylla/cassandra/lib/commons-lang3-3.1.jar:/usr/share/scylla/cassandra/
lib/commons-math3-3.2.jar:/usr/share/scylla/cassandra/lib/compress-lzf-0.8.4.jar
:/usr/share/scylla/cassandra/lib/concurrentlinkedhashmap-lru-1.4.jar:/usr/share/
scylla/cassandra/lib/concurrent-trees-2.4.0.jar:/usr/share/scylla/cassandra/lib/
disruptor-3.0.1.jar:/usr/share/scylla/cassandra/lib/ecj-4.4.2.jar:/usr/share/scy
lla/cassandra/lib/guava-18.0.jar:/usr/share/scylla/cassandra/lib/HdrHistogram-2.
1.9.jar:/usr/share/scylla/cassandra/lib/high-scale-lib-1.0.6.jar:/usr/share/scyl
la/cassandra/lib/hppc-0.5.4.jar:/usr/share/scylla/cassandra/lib/jackson-core-asl
-1.9.13.jar:/usr/share/scylla/cassandra/lib/jackson-mapper-asl-1.9.13.jar:/usr/s
hare/scylla/cassandra/lib/jamm-0.3.0.jar:/usr/share/scylla/cassandra/lib/javax.i
nject.jar:/usr/share/scylla/cassandra/lib/jbcrypt-0.3m.jar:/usr/share/scylla/cas
sandra/lib/jcl-over-slf4j-1.7.7.jar:/usr/share/scylla/cassandra/lib/jctools-core
-1.2.1.jar:/usr/share/scylla/cassandra/lib/jflex-1.6.0.jar:/usr/share/scylla/cas
sandra/lib/jna-4.2.2.jar:/usr/share/scylla/cassandra/lib/joda-time-2.4.jar:/usr/
share/scylla/cassandra/lib/json-simple-1.1.jar:/usr/share/scylla/cassandra/lib/j
stackjunit-0.0.1.jar:/usr/share/scylla/cassandra/lib/libthrift-0.9.2.jar:/usr/sh
are/scylla/cassandra/lib/log4j-over-slf4j-1.7.7.jar:/usr/share/scylla/cassandra/
lib/logback-classic-1.1.3.jar:/usr/share/scylla/cassandra/lib/logback-core-1.1.3
.jar:/usr/share/scylla/cassandra/lib/lz4-1.3.0.jar:/usr/share/scylla/cassandra/l
ib/metrics-core-3.1.5.jar:/usr/share/scylla/cassandra/lib/metrics-jvm-3.1.5.jar:
/usr/share/scylla/cassandra/lib/metrics-logback-3.1.5.jar:/usr/share/scylla/cass
andra/lib/netty-all-4.0.56.Final.jar:/usr/share/scylla/cassandra/lib/ohc-core-0.
4.4.jar:/usr/share/scylla/cassandra/lib/ohc-core-j8-0.4.4.jar:/usr/share/scylla/
cassandra/lib/reporter-config3-3.0.3.jar:/usr/share/scylla/cassandra/lib/reporte
r-config-base-3.0.3.jar:/usr/share/scylla/cassandra/lib/sigar-1.6.4.jar:/usr/sha
re/scylla/cassandra/lib/slf4j-api-1.7.7.jar:/usr/share/scylla/cassandra/lib/snak
eyaml-1.11.jar:/usr/share/scylla/cassandra/lib/snappy-java-1.1.1.7.jar:/usr/shar
e/scylla/cassandra/lib/snowball-stemmer-1.3.0.581.1.jar:/usr/share/scylla/cassan
dra/lib/ST4-4.0.8.jar:/usr/share/scylla/cassandra/lib/stream-2.5.2.jar:/usr/shar
e/scylla/cassandra/lib/thrift-server-0.3.7.jar:/usr/share/scylla/cassandra/apach
e-cassandra-3.11.3-SNAPSHOT.jar:/usr/share/scylla/cassandra/apache-cassandra.jar
:/usr/share/scylla/cassandra/apache-cassandra-thrift-3.11.3-SNAPSHOT.jar:/usr/sh
are/scylla/cassandra/scylla-tools-3.11.3-SNAPSHOT.jar:/usr/share/scylla/cassandr
a/stress.jar:

**Comment by wmac** [comment]
Also the dependency query gives the following:

[root@dane ~]# rpm -qaR scylla-tools-core
/bin/sh
/usr/bin/env
java-1.8.0-openjdk-headless
java-headless
python
python-yaml
rpmlib(CompressedFileNames) <= 3.0.4-1
rpmlib(FileDigests) <= 4.6.0-1
rpmlib(PayloadFilesHavePrefix) <= 4.0-1
rpmlib(PayloadIsXz) <= 5.2-1

**Comment by slivne** [comment]
I am not able to reproduce although I tried to setup the same env 

can you please run `rpm -qa`

and provide all the rpms maybe you have something else setup in parallel.

you env

```
/tmp/tmp.TwKLWhneft
/usr/share/scylla/cassandra/lib/airline-0.6.jar
/usr/share/scylla/cassandra/lib/antlr-runtime-3.5.2.jar
/usr/share/scylla/cassandra/lib/asm-5.0.4.jar
/usr/share/scylla/cassandra/lib/caffeine-2.2.6.jar
/usr/share/scylla/cassandra/lib/cassandra-driver-core-3.5.1-shaded.jar
/usr/share/scylla/cassandra/lib/commons-cli-1.1.jar
/usr/share/scylla/cassandra/lib/commons-codec-1.9.jar
/usr/share/scylla/cassandra/lib/commons-lang3-3.1.jar
/usr/share/scylla/cassandra/lib/commons-math3-3.2.jar
/usr/share/scylla/cassandra/lib/compress-lzf-0.8.4.jar
/usr/share/scylla/cassandra/lib/concurrentlinkedhashmap-lru-1.4.jar
/usr/share/scylla/cassandra/lib/concurrent-trees-2.4.0.jar
/usr/share/scylla/cassandra/lib/disruptor-3.0.1.jar
/usr/share/scylla/cassandra/lib/ecj-4.4.2.jar
/usr/share/scylla/cassandra/lib/guava-18.0.jar
/usr/share/scylla/cassandra/lib/HdrHistogram-2.1.9.jar
/usr/share/scylla/cassandra/lib/high-scale-lib-1.0.6.jar
/usr/share/scylla/cassandra/lib/hppc-0.5.4.jar
/usr/share/scylla/cassandra/lib/jackson-core-asl-1.9.13.jar
/usr/share/scylla/cassandra/lib/jackson-mapper-asl-1.9.13.jar
/usr/share/scylla/cassandra/lib/jamm-0.3.0.jar
/usr/share/scylla/cassandra/lib/javax.inject.jar
/usr/share/scylla/cassandra/lib/jbcrypt-0.3m.jar
/usr/share/scylla/cassandra/lib/jcl-over-slf4j-1.7.7.jar
/usr/share/scylla/cassandra/lib/jctools-core-1.2.1.jar
/usr/share/scylla/cassandra/lib/jflex-1.6.0.jar
/usr/share/scylla/cassandra/lib/jna-4.2.2.jar
/usr/share/scylla/cassandra/lib/joda-time-2.4.jar
/usr/share/scylla/cassandra/lib/json-simple-1.1.jar
/usr/share/scylla/cassandra/lib/jstackjunit-0.0.1.jar
/usr/share/scylla/cassandra/lib/libthrift-0.9.2.jar
/usr/share/scylla/cassandra/lib/log4j-over-slf4j-1.7.7.jar
/usr/share/scylla/cassandra/lib/logback-classic-1.1.3.jar
/usr/share/scylla/cassandra/lib/logback-core-1.1.3.jar
/usr/share/scylla/cassandra/lib/lz4-1.3.0.jar
/usr/share/scylla/cassandra/lib/metrics-core-3.1.5.jar
/usr/share/scylla/cassandra/lib/metrics-jvm-3.1.5.jar
/usr/share/scylla/cassandra/lib/metrics-logback-3.1.5.jar
/usr/share/scylla/cassandra/lib/netty-all-4.0.56.Final.jar
/usr/share/scylla/cassandra/lib/ohc-core-0.4.4.jar
/usr/share/scylla/cassandra/lib/ohc-core-j8-0.4.4.jar
/usr/share/scylla/cassandra/lib/reporter-config3-3.0.3.jar
/usr/share/scylla/cassandra/lib/reporter-config-base-3.0.3.jar
/usr/share/scylla/cassandra/lib/sigar-1.6.4.jar
/usr/share/scylla/cassandra/lib/slf4j-api-1.7.7.jar
/usr/share/scylla/cassandra/lib/snakeyaml-1.11.jar
/usr/share/scylla/cassandra/lib/snappy-java-1.1.1.7.jar
/usr/share/scylla/cassandra/lib/snowball-stemmer-1.3.0.581.1.jar
/usr/share/scylla/cassandra/lib/ST4-4.0.8.jar
/usr/share/scylla/cassandra/lib/stream-2.5.2.jar
/usr/share/scylla/cassandra/lib/thrift-server-0.3.7.jar
/usr/share/scylla/cassandra/apache-cassandra-3.11.3-SNAPSHOT.jar
/usr/share/scylla/cassandra/apache-cassandra.jar
/usr/share/scylla/cassandra/apache-cassandra-thrift-3.11.3-SNAPSHOT.jar
/usr/share/scylla/cassandra/scylla-tools-3.11.3-SNAPSHOT.jar
/usr/share/scylla/cassandra/stress.jar
```

trying to run on centos7.6 with the same jvm  I am getting the same envioronment

```
cat /etc/redhat-release 
CentOS Linux release 7.6.1810 (Core)
```

```
rpm -qa | grep java
javapackages-tools-3.4.1-11.el7.noarch
java-1.8.0-openjdk-headless-1.8.0.222.b10-1.el7_7.x86_64
python-javapackages-3.4.1-11.el7.noarch
tzdata-java-2019c-1.el7.noarch
```

```
/usr/share/scylla/cassandra/lib/airline-0.6.jar
/usr/share/scylla/cassandra/lib/antlr-runtime-3.5.2.jar
/usr/share/scylla/cassandra/lib/asm-5.0.4.jar
/usr/share/scylla/cassandra/lib/caffeine-2.2.6.jar
/usr/share/scylla/cassandra/lib/cassandra-driver-core-3.5.1-shaded.jar
/usr/share/scylla/cassandra/lib/commons-cli-1.1.jar
/usr/share/scylla/cassandra/lib/commons-codec-1.9.jar
/usr/share/scylla/cassandra/lib/commons-lang3-3.1.jar
/usr/share/scylla/cassandra/lib/commons-math3-3.2.jar
/usr/share/scylla/cassandra/lib/compress-lzf-0.8.4.jar
/usr/share/scylla/cassandra/lib/concurrentlinkedhashmap-lru-1.4.jar
/usr/share/scylla/cassandra/lib/concurrent-trees-2.4.0.jar
/usr/share/scylla/cassandra/lib/disruptor-3.0.1.jar
/usr/share/scylla/cassandra/lib/ecj-4.4.2.jar
/usr/share/scylla/cassandra/lib/guava-18.0.jar
/usr/share/scylla/cassandra/lib/HdrHistogram-2.1.9.jar
/usr/share/scylla/cassandra/lib/high-scale-lib-1.0.6.jar
/usr/share/scylla/cassandra/lib/hppc-0.5.4.jar
/usr/share/scylla/cassandra/lib/jackson-core-asl-1.9.13.jar
/usr/share/scylla/cassandra/lib/jackson-mapper-asl-1.9.13.jar
/usr/share/scylla/cassandra/lib/jamm-0.3.0.jar
/usr/share/scylla/cassandra/lib/javax.inject.jar
/usr/share/scylla/cassandra/lib/jbcrypt-0.3m.jar
/usr/share/scylla/cassandra/lib/jcl-over-slf4j-1.7.7.jar
/usr/share/scylla/cassandra/lib/jctools-core-1.2.1.jar
/usr/share/scylla/cassandra/lib/jflex-1.6.0.jar
/usr/share/scylla/cassandra/lib/jna-4.2.2.jar
/usr/share/scylla/cassandra/lib/joda-time-2.4.jar
/usr/share/scylla/cassandra/lib/json-simple-1.1.jar
/usr/share/scylla/cassandra/lib/jstackjunit-0.0.1.jar
/usr/share/scylla/cassandra/lib/libthrift-0.9.2.jar
/usr/share/scylla/cassandra/lib/log4j-over-slf4j-1.7.7.jar
/usr/share/scylla/cassandra/lib/logback-classic-1.1.3.jar
/usr/share/scylla/cassandra/lib/logback-core-1.1.3.jar
/usr/share/scylla/cassandra/lib/lz4-1.3.0.jar
/usr/share/scylla/cassandra/lib/metrics-core-3.1.5.jar
/usr/share/scylla/cassandra/lib/metrics-jvm-3.1.5.jar
/usr/share/scylla/cassandra/lib/metrics-logback-3.1.5.jar
/usr/share/scylla/cassandra/lib/netty-all-4.0.56.Final.jar
/usr/share/scylla/cassandra/lib/ohc-core-0.4.4.jar
/usr/share/scylla/cassandra/lib/ohc-core-j8-0.4.4.jar
/usr/share/scylla/cassandra/lib/reporter-config3-3.0.3.jar
/usr/share/scylla/cassandra/lib/reporter-config-base-3.0.3.jar
/usr/share/scylla/cassandra/lib/sigar-1.6.4.jar
/usr/share/scylla/cassandra/lib/slf4j-api-1.7.7.jar
/usr/share/scylla/cassandra/lib/snakeyaml-1.11.jar
/usr/share/scylla/cassandra/lib/snappy-java-1.1.1.7.jar
/usr/share/scylla/cassandra/lib/snowball-stemmer-1.3.0.581.1.jar
/usr/share/scylla/cassandra/lib/ST4-4.0.8.jar
/usr/share/scylla/cassandra/lib/stream-2.5.2.jar
/usr/share/scylla/cassandra/lib/thrift-server-0.3.7.jar
/usr/share/scylla/cassandra/apache-cassandra-3.11.3-SNAPSHOT.jar
/usr/share/scylla/cassandra/apache-cassandra.jar
/usr/share/scylla/cassandra/apache-cassandra-thrift-3.11.3-SNAPSHOT.jar
/usr/share/scylla/cassandra/scylla-tools-3.11.3-SNAPSHOT.jar
/usr/share/scylla/cassandra/stress.jar
```

and it works

```
nodetool status
Datacenter: datacenter1
=======================
Status=Up/Down
|/ State=Normal/Leaving/Joining/Moving
--  Address    Load       Tokens       Owns    Host ID                               Rack
UN  127.0.0.1  361.68 KB  256          ?       0336413e-f7b5-47f1-a28d-c8eb3d3cd25f  rack1

Note: Non-system keyspaces don't have the same replication settings, effective ownership information is meaningless
```

**Comment by slivne** [comment]
stale issue - closing it.
@wmac if you could not resolve it - please post and we will reopen

**Comment by pierrediancourt** [comment]
Hi,

I had @wmac symptoms in this context :
```
cat /etc/redhat-release
Rocky Linux release 8.9 (Green Obsidian)

dnf list installed scylla*
scylla-enterprise.x86_64                 2022.1.14-0.20240123.e90d107d48
scylla-enterprise-conf.x86_64            2022.1.14-0.20240123.e90d107d48
scylla-enterprise-jmx.noarch             2022.1.14-0.20240123.e90d107d48
scylla-enterprise-kernel-conf.x86_64     2022.1.14-0.20240123.e90d107d48
scylla-enterprise-node-exporter.x86_64   2022.1.14-0.20240123.e90d107d48
scylla-enterprise-python3.x86_64         2022.1.5-0.20230108.8c2c21866  
scylla-enterprise-server.x86_64          2022.1.14-0.20240123.e90d107d48
scylla-enterprise-tools.noarch           2022.1.14-0.20240123.e90d107d48
scylla-enterprise-tools-core.noarch      2022.1.14-0.20240123.e90d107d48
scylla-manager-client.x86_64             3.2.8~0.20240517.5f324acd2-1
scylla-manager-server.x86_64             3.2.8~0.20240517.5f324acd2-1

rpm -qa | grep java
java-1.8.0-openjdk-headless-1.8.0.412.b08-2.el8.x86_64
javapackages-filesystem-5.3.0-2.module+el8.3.0+125+5da1ae29.noarch
tzdata-java-2024a-1.el8.noarch
```

Running `systemctl restart scylla-jmx` fixed it

**Comment by mykaul** [comment]
@pierrediancourt - if you are using an enterprise release, please open a support case. Also, since 2022.1 we've released 2022.2, 2023.1 and 2024.1, offering hundreds or more fixes and improvements - so upgrade is encouraged.
